### PR TITLE
[type: feature]Add RpcContextPlugin to transmit header to rpc context.

### PR DESCRIPTION
**Pull request by KevinClair**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Review by yu199195 — CHANGES_REQUESTED** [review]
wait me review

File: shenyu-plugin/shenyu-plugin-rpc-context/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>shenyu-plugin</artifactId>
+        <groupId>org.apache.shenyu</groupId>
+        <version>2.4.2-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>shenyu-plugin-rpc-context</artifactId>
```
Comment (tydhot) [inline shenyu-plugin/shenyu-plugin-rpc-context/pom.xml:29]: The name rpccontext is the same as Dubbo's rpccontext, which is easy to be misunderstood

Comment (KevinClair) [reply shenyu-plugin/shenyu-plugin-rpc-context/pom.xml:29]: Yes, it was originally to pass Dubbo's rpccontext, so I named the plug with rpccontext. Or you can suggest a better name to explain the meaning of the plug-in.

Comment (tydhot) [reply shenyu-plugin/shenyu-plugin-rpc-context/pom.xml:29]: Why not make a general context? The client plugin obtains and injects its own implementation from this context, and the server plugin obtains and writes the implementation of this context from its own request.After all, besides Dubbo, grpc also has its own context to pass

Comment (KevinClair) [reply shenyu-plugin/shenyu-plugin-rpc-context/pom.xml:29]: Sounds great, how about rename to `shenyu-plugin-general-context`?Or maybe just `shenyu-plugin-context`?

**Comment by KevinClair** [comment]
Done, dashboard see [this](https://github.com/apache/incubator-shenyu-dashboard/pull/165)

**Review by yu199195 — APPROVED** [review]

**Pull request merged**